cakeshop-integration-test/src/test/java/uk/gov/justice/services/cakeshop/it/RebuildSnapshotsIT.java

```diff
@@ -0,0 +1,227 @@
+import uk.gov.justice.services.common.util.UtcClock;
+import uk.gov.justice.services.eventsourcing.jdbc.snapshot.SnapshotJdbcRepository;
+import uk.gov.justice.services.eventsourcing.jdbc.snapshot.StandaloneSnapshotJdbcRepositoryFactory;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.event.Event;
+import uk.gov.justice.services.eventsourcing.util.jee.timer.StopWatchFactory;
+import uk.gov.justice.services.jmx.api.domain.CommandState;
+import uk.gov.justice.services.jmx.api.domain.SystemCommandStatus;
+import uk.gov.justice.services.jmx.api.mbean.SystemCommanderMBean;
+import uk.gov.justice.services.jmx.api.parameters.JmxCommandRuntimeParameters;
+import uk.gov.justice.services.jmx.system.command.client.SystemCommanderClient;
+import uk.gov.justice.services.jmx.system.command.client.TestSystemCommanderClientFactory;
+import uk.gov.justice.services.messaging.Metadata;
+import uk.gov.justice.services.test.utils.core.messaging.Poller;
+import uk.gov.justice.services.test.utils.persistence.DatabaseCleaner;
+
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.json.JsonArrayBuilder;
+import javax.sql.DataSource;
+
+import org.apache.commons.lang3.time.StopWatch;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RebuildSnapshotsIT {
+
+    private static final String EVENT_SOURCE = "cakeshop";
+    private static final int NUMBER_OF_EVENTS_ON_STREAM_TO_CREATE = 100_000;
+
+    private final TestSystemCommanderClientFactory systemCommanderClientFactory = new TestSystemCommanderClientFactory();
+    private final DatabaseCleaner databaseCleaner = new DatabaseCleaner();
+    private final UtcClock clock = new UtcClock();
+    private final DataSource eventStoreDataSource = new DatabaseManager().initEventStoreDb();
+    private final SnapshotJdbcRepository snapshotJdbcRepository = new StandaloneSnapshotJdbcRepositoryFactory().getSnapshotJdbcRepository(eventStoreDataSource);
+    private final BatchEventInserter batchEventInserter = new BatchEventInserter(eventStoreDataSource, 10_000);
+    private final Poller longPoller = new Poller(2400, 1000L);
+
+    @BeforeEach
+    public void cleanDatabase() {
+        databaseCleaner.cleanEventStoreTables(DB_CONTEXT_NAME);
+        cleanViewstoreTables();
+        databaseCleaner.cleanSystemTables(DB_CONTEXT_NAME);
+    }
+
+    @Test
+    public void shouldUseJmxCommandToHydrateAggregateAndStoreSnapshot() throws Exception {
+
+        final String aggregateClassName = Recipe.class.getName();
+        final UUID recipeId = randomUUID();
+
+        snapshotJdbcRepository.removeAllSnapshots(recipeId, Recipe.class);
+        assertThat(snapshotJdbcRepository.getLatestSnapshot(recipeId, Recipe.class), is(empty()));
+
+        System.out.println("Generating " + NUMBER_OF_EVENTS_ON_STREAM_TO_CREATE + " events");
```

Comment:
maybe logger?

Reply:
It's very noisy to use a logger. Also, as the wildfly logging is sent to the console during cake-shop integration tests, it's hard to then see if the log message is a wildfly log message, or just information outputted by the test